server/src/test/java/io/crate/planner/consumer/GroupByPlannerTest.java

```diff
@@ -711,4 +707,30 @@ public void testGroupByOrderByPartitionedClolumn() throws Exception {
         assertThat(orderBy, instanceOf(InputColumn.class));
         assertThat(orderBy.valueType(), is(DataTypes.TIMESTAMPZ));
     }
+
+    @Test
+    public void test_group_by_constant_null() throws IOException {
+        var e = SQLExecutor.builder(clusterService, 2, RandomizedTest.getRandom(), List.of())
+            .addTable(TableDefinitions.USER_TABLE_DEFINITION)
+            .build();
+        Merge merge = e.plan("select nn, sum(ints) \n" +
+            "from (select null as nn, ints from users) t \n" +
+            "group by nn");
+        Merge subMerge = (Merge) merge.subPlan();
+        Collect collect  = (Collect) subMerge.subPlan();
+        assertThat(collect.collectPhase().toCollect().get(0),  isLiteral(null));
+        assertThat(collect.collectPhase().toCollect().get(1),  isReference("ints"));
+    }
+
+    @Test
+    public void test_group_by_of_constant_aliased_null() throws IOException {
+        var e = SQLExecutor.builder(clusterService, 2, RandomizedTest.getRandom(), List.of())
+            .addTable(TableDefinitions.USER_TABLE_DEFINITION)
+            .build();
+        Merge merge = e.plan("select null as nn, sum(ints) from users group by nn");
+        Merge subMerge = (Merge) merge.subPlan();
+        Collect collect  = (Collect) subMerge.subPlan();
+        assertThat(collect.collectPhase().toCollect().get(1), isLiteral(null));
```

Comment:
I thought I need `contaisInAnyOrder` - but looks like Reference is always first to collect here. In the prev test it's other way around. I have an integration test verifying that result is correct.